src/extended_json.ts

```diff
@@ -70,17 +72,23 @@ const keysToCodecs = {
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 function deserializeValue(value: any, options: EJSONOptions = {}) {
   if (typeof value === 'number') {
+    const in32BitRange = value <= BSON_INT32_MAX && value >= BSON_INT32_MIN;
+    const in64BitRange = value <= BSON_INT64_MAX && value >= BSON_INT64_MIN;
+
     if (options.relaxed || options.legacy) {
-      return value;
+      return Number.isInteger(value) && options.useBigInt64 ? BigInt(value) : value;
     }
 
     if (Number.isInteger(value) && !Object.is(value, -0)) {
       // interpret as being of the smallest BSON integer type that can represent the number exactly
-      if (value >= BSON_INT32_MIN && value <= BSON_INT32_MAX) {
+      if (in32BitRange) {
         return new Int32(value);
       }
-      if (value >= BSON_INT64_MIN && value <= BSON_INT64_MAX) {
+      if (in64BitRange) {
         // TODO(NODE-4377): EJSON js number handling diverges from BSON
```

Comment:
This TODO was related to the int64BitRange boolean logic, can you move it up to be with that variable

Reply:
@W-A-James bump on this